core/src/main/scala/org/apache/spark/sql/delta/commands/OptimizeTableCommand.scala

```diff
@@ -125,14 +125,16 @@ case class OptimizeTableCommand(
     val partitionColumns = txn.snapshot.metadata.partitionColumns
     // Parse the predicate expression into Catalyst expression and verify only simple filters
     // on partition columns are present
-    val partitionPredicates = partitionPredicate.map(predicate => {
+
+    val partitionPredicates = userPartitionPredicates.flatMap(partitionPredicate =>
```

Comment:
nit: maybe re-format the code a bit like this:
```
    val partitionPredicates = userPartitionPredicates.flatMap { partitionPredicate =>
      partitionPredicate.map { predicate =>
        val predicates = parsePredicates(sparkSession, predicate)
        verifyPartitionPredicates(
          sparkSession,
          partitionColumns,
          predicates)
        predicates
      }.getOrElse(Seq.empty)
    }
```
The current format took me a while to realize `getOrElse` is inside the function.